dashboard/test/models/sections/section_test.rb

```diff
@@ -585,10 +585,11 @@ def verify(actual, expected)
     end
     group_1_name = 'new_group_1'
     group_2_name = 'new_group_2'
-    new_groups = [
-      {name: group_1_name, members: [{follower_id: followers[0].id}]},
-      {name: group_2_name, members: [{follower_id: followers[2].id}, {follower_id: followers[3].id}]}
-    ]
+    # Parameters are represented as hashes
+    new_groups = {
+      '0' => {name: group_1_name, members: {'0' => {follower_id: followers[0].id}}},
```

Comment:
This feels unusual to me...I think using jQuery's post isn't super common for us either (only seeing a handful of uses in our code base) -- is there a reason to not use something like this? Personal preference?

https://github.com/code-dot-org/code-dot-org/blob/7387139155b277f071e08a3a9719fe77debfd0ac/apps/src/code-studio/pd/foorm/editor/components/FoormEntityEditorHeader.jsx#L55

Reply:
I believe this is due to how Rails's action controller parameters are represented. The data that is POST-ed is still a JSON object with arrays, i.e.
```
{
   groups: [
      { id: 1, members: [...] },
      { id: 2, members: [...] },
      ...
   ]
}
```

But from what I was seeing, in Rails, the parameters were converted to hashes, with indices being treated as keys, i.e.
```
{
   groups: {
     0: { id: 1, members: { 0: {...}, 1: {...} },
     1: { id: 2, members: {...} },
      ...
   }
}
```
Let me double check if the jQuery method makes a difference though. AFAIK, `$.post` is just shorthand for `$.ajax({ type: "POST", url: ... })`

Reply:
Maybe something like this? I do see content type `application/json` often:

https://stackoverflow.com/questions/6410810/rails-not-decoding-json-from-jquery-correctly-array-becoming-a-hash-with-intege

Reply:
Actually looking online, you're right :) I think because `contentType: application/json` isn't being set, it's being converted to this weird hash format. Let me update this and confirm.